### PR TITLE
Updates servicing doc and servicing template

### LINKED ISSUE #39646: Add PR templates (default and servicing)
As we approach .NET 5, we are nearing branching off .NET 5 and moving to a servicing model after release. For the servicing model, we have an ad-hoc PR template that we require, but we don't have it saved anywhere. We end up pointing people to previous servicing PRs as examples.

We should add a servicing PR template to simplify this process and make it easier to set up servicing PRs right the first time.

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/39646

https://github.com/Anipik/runtime/compare/master...template?expand=1&template=servicing_pull_request_template.md

Constraints:-
- The template needs to be in default branch under pull_request_template folder.
- In case of multiple templates, you cannot directly  add the default template to the prs. you need to do that through a url.
- The link in the doc will be working after we merge the pr (i added an example in the description for my fork)

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+<!-- Describe the impact of the issue as well as the pull request. -->
+
+# Regression
+
+<!-- Mention if its  a new bug or a regression. -->
+
+# Testing
+
+<!-- What kiind of testing has been done with the fix. -->
```
Comment (safern) [inline PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md:19]:
```suggestion
<!-- What kind of testing has been done with the fix. -->
```

**Review by safern — COMMENTED** [review]
Should we mention the `/backport` action somewhere?

**Comment by Anipik** [comment]
> Should we mention the /backport action somewhere?

the branches have already diverged that it is causing merge conflicts.

**Comment by safern** [comment]
> the branches have already diverged that it is causing merge conflicts.

What do you mean?

I meant if we should mention the /backport comment available to backport a PR to a servicing branch in the docs?

**Comment by Anipik** [comment]
backporting wont help us much in servicing because master branch would have diverged a lot from the servicing branches and backporting dont work with merge conflicts.
The other reason is the person still will have to add the commit for  packageVersion and assemblyVersion stuff. I dont think all developers have push permission access. Access will also refrain from the developers from making any change in their backporting pr.

**Comment by safern** [comment]
Ah I see. you're right. Makes sense.

File: docs/project/library-servicing.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
-For example, if you made changes to [System.Data.SqlClient](https://github.com/dotnet/runtime/tree/master/src/libraries/Microsoft.Win32.Registry), then you have a .pkgproj, and will have to follow the steps in this document. However, if you made changes to [System.Collections](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Collections), then you don't have a .pkgproj, and you do not need to do any further work for servicing.
+For example, if you made changes to [Microsoft.Win32.Primitives](https://github.com/dotnet/runtime/tree/master/src/libraries/Microsoft.Win32.Primitives), then you have a .pkgproj, and will have to follow the steps in this document. However, if you made changes to [System.Collections](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Collections), then you don't have a .pkgproj, and you do not need to do any further work for servicing.
```
Comment (safern) [inline docs/project/library-servicing.md:9]:
```suggestion
For example, if you made changes to [Microsoft.Win32.Primitives](https://github.com/dotnet/runtime/tree/master/src/libraries/Microsoft.Win32.Primitives), then you have a `.pkgproj`, and will have to follow the steps in this document. However, if you made changes to [System.Collections](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Collections), then you don't have a .pkgproj, and you do not need to do any further work for servicing.
```

**Review by safern — APPROVED** [review]

File: PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Master PR <!-- PR Number if the fix has already been merged in master branch. -->
+
+# Description
+
+<!-- Give a brief summary of the issue and how the pull request is fixing it. -->
+
+# Customer Impact
+
+<!-- Describe the impact of the issue as well as the pull request. -->
```
Comment (danmosemsft) [inline PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md:11]:
```suggestion
<!-- What is the impact to customers of not taking this fix? -->
```

**Comment by danmosemsft** [comment]
I guess we should add ` PULL_REQUEST_TEMPLATE/*` to the list of folders that should not trigger CI.

Does that mean an entry here? https://github.com/dotnet/runtime/blob/3cf050a9e4a9f5c79116c2a3018eb61c417df574/eng/pipelines/runtime.yml#L37

File: PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+Fixes Issue <!-- Issue Number -->
+
+Master PR <!-- PR Number if the fix has already been merged in master branch. -->
```
Comment (danmosemsft) [inline PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md:3]:
```suggestion
Master PR <!-- Link to PR if any that fixed this in the master branch. -->
```

File: PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+<!-- Give a brief summary of the issue and how the pull request is fixing it. -->
+
+# Customer Impact
+
+<!-- Describe the impact of the issue as well as the pull request. -->
+
+# Regression
+
+<!-- Mention if its  a new bug or a regression. -->
```
Comment (danmosemsft) [inline PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md:15]:
```suggestion
<!-- Is this fixing a problem that was introduced in the most recent release, ie., fixing a regression? -->
```

File: PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+# Testing
+
+<!-- What kind of testing has been done with the fix. -->
+
+# Risk
+
+<!-- Risks associated with the change. -->
```
Comment (danmosemsft) [inline PULL_REQUEST_TEMPLATE/servicing_pull_request_template.md:23]:
```suggestion
<!-- Please assess the risk of taking this fix. Provide details backing up your assessment. -->
```

File: docs/project/library-servicing.md [hunk 7]
```diff
@@ -47,3 +47,7 @@ All that's left is to ensure that your changes have worked as expected. To do so
 
 3. Try installing the built package in a test application, testing that your changes to the library are present & working as expected.
    To install your package add your local packages folder as a feed source in VS or your nuget.config and then add a PackageReference to the specific version of the package you built then try using the APIs.
+
+## Approval Process
+
+All the servicing changes needs to go through approval process. You should fill this [template](https://github.com/dotnet/runtime/compare/template?expand=1&template=servicing_pull_request_template.md) and add in the descripition of your pull request. You should also add `servicing-consider` label to the pull request.
```
Comment (danmosemsft) [inline docs/project/library-servicing.md:53]:
```suggestion
All the servicing changes needs to go through approval process. You should fill this [template](https://github.com/dotnet/runtime/compare/template?expand=1&template=servicing_pull_request_template.md) and add in the descripition of your pull request. You should also add `servicing-consider` label to the pull request and bring it to the attention of the engineering lead responsible for the area.
```

**Review by danmoseley — APPROVED** [review]
Can't review the details of servicing but the template looks fine to me..

**Comment by MichalStrehovsky** [comment]
> The template needs to be in default branch under pull_request_template folder.

Why can't this be under the `.github` folder, like the issue template?

File: eng/pipelines/runtime.yml [hunk 8]
```diff
@@ -17,6 +17,8 @@ trigger:
     - eng/Version.Details.xml
     - .github/*
     - docs/*
+    - ISSUE_TEMPLATE/*
```
Comment (safern) [inline eng/pipelines/runtime.yml:20]: We need to add these to `runtime-official.yml` as well.

**Comment by ViktorHofer** [comment]
ping @Anipik regarding

> Why can't this be under the .github folder, like the issue template?

**Comment by Anipik** [comment]
yes i will move it to .github

**Pull request merged**